samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -152,32 +148,26 @@ public void setUp() {
     for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
       LOGGER.info("Creating kafka topic: {}.", kafkaTopic);
       TestUtils.createTopic(zkUtils(), kafkaTopic, 5, 1, servers(), new Properties());
+      if (AdminUtils.topicExists(zkUtils(), kafkaTopic)) {
+        LOGGER.info("Topic: {} was created", kafkaTopic);
+      } else {
+        Assert.fail(String.format("Unable to create kafka topic: %s.", kafkaTopic));
+      }
     }
     for (String kafkaTopic : ImmutableList.of(inputSinglePartitionKafkaTopic, outputSinglePartitionKafkaTopic)) {
       LOGGER.info("Creating kafka topic: {}.", kafkaTopic);
       TestUtils.createTopic(zkUtils(), kafkaTopic, 1, 1, servers(), new Properties());
-    }
-  }
-
-  //  @Override
-  public void tearDown() {
-    if (zookeeper().zookeeper().isRunning()) {
-      for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
-        LOGGER.info("Deleting kafka topic: {}.", kafkaTopic);
-        AdminUtils.deleteTopic(zkUtils(), kafkaTopic);
-      }
-      for (String kafkaTopic : ImmutableList.of(inputSinglePartitionKafkaTopic, outputSinglePartitionKafkaTopic)) {
-        LOGGER.info("Deleting kafka topic: {}.", kafkaTopic);
-        AdminUtils.deleteTopic(zkUtils(), kafkaTopic);
+      if (AdminUtils.topicExists(zkUtils(), kafkaTopic)) {
+        LOGGER.info("Topic: {} was created", kafkaTopic);
+      } else {
+        Assert.fail(String.format("Unable to create kafka topic: %s.", kafkaTopic));
       }
-      zkUtils.close();
-      super.tearDown();
```

Comment:
Did you remove zk.close() completely?

Reply:
This was unintentionally removed. Adding the whole method back.